release/release_workflow.wdl

```diff
@@ -0,0 +1,280 @@
+   Boolean dryRun = false
+   
+   # Can be swapped out to try this on a fork
+   String organization = "broadinstitute"
+    
+   command {
+     set -e
+     set -x 
+     
+     # Clone repo and checkout develop
+     git clone https://github.com/${organization}/${repo}.git
+     cd ${repo}
+     git checkout develop
+     git pull --rebase
+
+     # Extract current version  
+     currentV=$(grep "git.baseVersion " build.sbt | cut -d "=" -f 2 | tr -d \"\,[:space:])
+
+     echo "Updating ${repo} to ${releaseV}"
+     ${updateVersionForCurrentRelaseCommand}
+     
+     echo "Update dependencies"
+     ${sep='\n' dependencyCommands}
+     
+     # Make sure it compiles
+     sbt update
+     sbt compile
+     
+     git add .
+     # If there is nothing to commit, git commit will return 1 which will fail the script.
+     # This ensures we only commit if build.sbt was effectively updated
+     git diff-index --quiet HEAD || git commit -m "Update ${repo} version from $currentV to ${releaseV}"
+       
+     # wdl4s needs a scala docs update
+     if [ ${repo} == "wdl4s" ]; then
+       sbt 'set scalacOptions in (Compile, doc) := List("-skip-packages", "better")' doc
+       git checkout gh-pages
+       mv target/scala-2.11/api ${releaseV}
+       git add ${releaseV}
+       git commit -m "API Docs"
+       git push ${true="--dry-run" false="" dryRun} origin gh-pages
+       git checkout develop
+       sed -i '' s/$currentV/${releaseV}/g README.md
+       git add README.md
+       git diff-index --quiet HEAD || git commit -m "Update ${repo} API docs references"
+     fi
+     
+     # Merge develop into master
+     git checkout master
+     git pull --rebase
+     git merge develop
+     
+     # Tag the release
+     git tag ${releaseV}
+     
+     # Push master and push the tags
+     git push ${true="--dry-run" false="" dryRun} origin master
+     git push ${true="--dry-run" false="" dryRun} --tags
+     
+     # Assemble jar if required
+     ${default = "" assembleJarCommand}
+     pwd > executionDir.txt
+     
+     # Update build.sbt on develop to point to next release version
+     git checkout develop
+     ${updateVersionForNextRelaseCommand}
+     git add .
+     git diff-index --quiet HEAD || git commit -m "Update ${repo} version from ${releaseV} to ${nextReleaseV}"
+     git push ${true="--dry-run" false="" dryRun} origin develop
+     
+     # Create and push the hotfix branch
+     git checkout -b ${releaseV}_hotfix ${releaseV}
+     git push origin ${releaseV}_hotfix
+   }
+
+   output {
+     String version = releaseV
+     String executionDir = read_string(repo + "/executionDir.txt")
+   }
+}
+
+task wait_for_artifactory {
+    String repo
+    String version
+    
+    command <<<
+        checkIfPresent() {
+            isPresent=$(curl -s --head https://artifactory.broadinstitute.org/artifactory/simple/libs-release-local/org/broadinstitute/${repo}/${version}/ | head -n 1 | grep "HTTP/1.[01] [23].." > /dev/null)
```

Comment:
POSIX `grep`s have `-q` for this:

http://unix.stackexchange.com/questions/104641/how-can-i-suppress-output-from-grep-so-that-it-only-returns-the-exit-status